src/components/dropdown.tsx

```diff
@@ -0,0 +1,144 @@
+import React, { useState, useEffect } from "react";
+import { clsx } from "clsx";
+
+import "./dropdown.scss";
+
+interface IOption {
+  name: string;
+  value?: string;
+}
+
+interface IDropdownProps<T extends IOption> {
+  value: string;
+  options: T[];
+  onSelect: (place: T) => void;
+  onSearchChange?: (value: string) => void;
+  label: string;
+  inputPlaceholder?: string;
+  icon?: React.ReactNode;
+  inline?: boolean;
+  width?: string;
+}
+
+export const Dropdown = <T extends IOption>({
+  value,
+  options,
+  onSelect,
+  onSearchChange,
+  label,
+  inputPlaceholder,
+  icon,
+  inline,
+  width
+}: IDropdownProps<T>) => {
+  const [showOptions, setShowOptions] = useState(false);
+  const [focusedOptionIndex, setFocusedOptionIndex] = useState(-1);
+
+  useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      if (event.target instanceof HTMLElement) {
+        if (!event.target.closest(".day-length-dropdown")) {
+          setShowOptions(false);
+        }
+      }
+    };
+
+    document.addEventListener("click", handleClickOutside);
+    return () => {
+      document.removeEventListener("click", handleClickOutside);
+    };
```

Comment:
Click events are sent fairly late in the life cycle of a mouse event, which leaves lots of opportunity for other code to interfere, e.g. via `preventDefault()`. Most attempts to handle this case try to intercept the event as early as possible, e.g. by listening to pointer events (which precede touch or mouse events) and passing "true" for capture so that the intercept occurs early. That said, I see that this code just moved here from elsewhere and so predates this PR, so I leave it to your discretion whether to change anything at this point.